src/providers/wfs/oapif/qgsoapifitemsrequest.cpp

```diff
@@ -150,74 +163,160 @@ void QgsOapifItemsRequest::processReply()
+              {
+                mFoundIdInProperties = true;
+              }
             }
           }
         }
       }
-    }
 
-    const auto links = QgsOAPIFJson::parseLinks( j );
-    mNextUrl = QgsOAPIFJson::findLink( links, QStringLiteral( "next" ), { QStringLiteral( "application/geo+json" ) } );
+      const auto links = QgsOAPIFJson::parseLinks( j );
+      mNextUrl = QgsOAPIFJson::findLink( links, QStringLiteral( "next" ), { QStringLiteral( "application/geo+json" ) } );
 
-    if ( j.is_object() && j.contains( "numberMatched" ) )
-    {
-      const auto numberMatched = j["numberMatched"];
-      if ( numberMatched.is_number_integer() )
+      if ( j.is_object() && j.contains( "numberMatched" ) )
       {
-        mNumberMatched = numberMatched.get<int>();
+        const auto numberMatched = j["numberMatched"];
+        if ( numberMatched.is_number_integer() )
+        {
+          mNumberMatched = numberMatched.get<int>();
+        }
       }
     }
+    catch ( const json::parse_error &ex )
+    {
+      mErrorCode = QgsBaseNetworkRequest::ApplicationLevelError;
+      mAppLevelError = ApplicationLevelError::JsonError;
+      mErrorMessage = errorMessageWithReason( tr( "Cannot decode JSON document: %1" ).arg( QString::fromStdString( ex.what() ) ) );
+      emit gotResponse();
+      return;
+    }
   }
-  catch ( const json::parse_error &ex )
+  else
   {
-    mErrorCode = QgsBaseNetworkRequest::ApplicationLevelError;
-    mAppLevelError = ApplicationLevelError::JsonError;
-    mErrorMessage = errorMessageWithReason( tr( "Cannot decode JSON document: %1" ).arg( QString::fromStdString( ex.what() ) ) );
-    emit gotResponse();
-    return;
+    // Get numberMatched and next link from HTTP response headers
+    for ( const auto &headerKeyValue : mResponseHeaders )
+    {
+      if ( headerKeyValue.first.compare( QByteArray( "OGC-NumberMatched" ), Qt::CaseSensitivity::CaseInsensitive ) == 0 )
+      {
+        bool ok = false;
+        const int val = QString::fromUtf8( headerKeyValue.second ).toInt( &ok );
+        if ( ok )
+          mNumberMatched = val;
+      }
+      else if ( headerKeyValue.first.compare( QByteArray( "Link" ), Qt::CaseSensitivity::CaseInsensitive ) == 0 )
+      {
+        // Parse stuff like:
+        //  <https://ogc-api.nrw.de/lika/v1/collections/flurstueck/items?f=html>; rel="alternate"; title="This document as HTML"; type="text/html", <https://ogc-api.nrw.de/lika/v1/collections/flurstueck/items?f=fgb&offset=10>; rel="next"; title="Next page"; type="application/flatgeobuf"
+
+        // Split on commas, except when they are in double quotes or between <...>, and skip padding space before/after separator
+        const thread_local QRegularExpression splitOnComma( R"(\s*,\s*(?=(?:[^"<]|"[^"]*"|<[^>]*>)*$))" );
+        const QStringList links = QString::fromUtf8( headerKeyValue.second ).split( splitOnComma );
+        for ( const QString &link : links )
```

Comment:
```suggestion
        for ( const QString &link : std::as_const( links ) )
```
Redudndant, I know but I was instructed to do so by top QGIS core developers and I agree that it makes the intent clearer.